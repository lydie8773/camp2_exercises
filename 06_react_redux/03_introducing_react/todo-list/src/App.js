import React, { Component } from 'react';
import logo from './logo.svg';
import './App.css';
import TodoForm from "./TodoForm";

class App extends Component {
  render() {
    return (
      <div className="App">
        <header className="App-header">
          <img src={logo} className="App-logo" alt="logo" />
          <h1 className="App-title">Todo list</h1>
        </header>
        <p className="App-intro">
					<div class="container">
						<TodoForm/>
					</div>
        </p>
      </div>
    );
  }
}

export default App;
