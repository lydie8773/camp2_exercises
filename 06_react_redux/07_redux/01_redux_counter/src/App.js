// import React, { Component } from 'react';
//
// import './App.css';
// import Counter from "./Counter";
// import { connect } from "react-redux";
//
//
// class App extends Component {
//
//   onIncrement() {
//     this.props.Increase();
//   }
//
//   onDecrement() {
//     this.props.Decrease();
//   }
//   render() {
//
//     return (
//       <div>
//         <h2>Counter</h2>
//         <p>The counter value is {this.props.count}</p>
//         <button onClick={this.onIncrement.bind(this)}>Increase</button>
//         <button onClick={this.onDecrement.bind(this)}>Decrease</button>
//       </div>
//     )
//   }
//
//
//
//   // render() {
//   //   return (
//   //     <div className="App">
//   //       <header className="App-header">
//   //         <img src={logo} className="App-logo" alt="logo" />
//   //         <h1 className="App-title">Welcome to React</h1>
//   //       </header>
//   //       <p className="App-intro">
//   //         {/*<Counter value={store.getState()}/>*/}
//   //         <Counter/>
//   //       </p>
//   //     </div>
//   //   );
//   // }
// }
//
// const mapStateToProps = function (state) {
//   return state;
// };
//
// const mapDispatchToProps = function (dispatch) {
//   return {
//     Increase: function () {
//       return dispatch({type: "COUNTER_INCRE"})
//     },
//     Decrease: function () {
//       return dispatch({type: "COUNTER_DECRE"})
//     }
//   }
// };
//
// export const CounterPackage = connect(mapStateToProps, mapDispatchToProps)(Counter);
//
//
// export default App;







import React, {Component} from 'react';
import {connect} from 'react-redux';

class App extends Component {

  onIncrement() {
    this.props.Increase();
  }

  onDecrement() {
    this.props.Decrease();
  }

  render() {

    console.log(this.props.count);
    return (
      <div>
        <h2>Counter</h2>
        <p>The counter value is {this.props.count}</p>
        <button onClick={this.onIncrement.bind(this)}>+</button>
        <button onClick={this.onDecrement.bind(this)}>-</button>
      </div>
    )
  }
}
const mapStateToProps = function (state) {
  return { count: state.count }
};

const mapDispatchToProps = function (dispatch) {
  return {
    Increase: function () {
      return dispatch({
        type: "COUNTER_INCRE"
      })
    },
    Decrease: function () {
      return dispatch({
        type: "COUNTER_DECRE"
      })
    }
  }
};

const ConnectedApp = connect(mapStateToProps, mapDispatchToProps)(App);


export default ConnectedApp;

