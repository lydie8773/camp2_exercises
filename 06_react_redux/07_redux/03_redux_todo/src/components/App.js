import React from 'react';

import TodoList from '../containers/TodoList';
import TodoInput from '../containers/TodoInput';

const App = () => {
  return(
    <div className="container">
      <h1 className="text-center">Your Todo List</h1>
      <TodoInput />
      <TodoList />

    </div>
  );
}

export default App;
