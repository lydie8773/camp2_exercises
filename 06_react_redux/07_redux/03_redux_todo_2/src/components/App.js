import React from 'react';

import Layout from './Layout';
import TodoList from '../containers/TodoList';
import TodoInput from '../containers/TodoInput';

const App = () => {
  return(
    <Layout>
      <TodoList />
      <TodoInput />
    </Layout>
  );
}

export default App;
