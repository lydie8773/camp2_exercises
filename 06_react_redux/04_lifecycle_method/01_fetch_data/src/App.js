import React, { Component } from 'react';
// import logo from './logo.svg';
import './App.css';





class DogApp extends Component {
	constructor(props) {
		super(props);
		this.state = {
			urlPic: ""
		};
		this.componentDidMount = this.componentDidMount.bind(this);
	}

	componentDidMount (){
		fetch("https://random.dog/woof.json", {
			method: "GET"
		})
			.then((response) => response.json())
			.then((data) => {
					console.log("data", data.url);
					this.setState({
						urlPic: data.url
					})
			})
	}


  render() {
    return (
      <div className="App">
				<button onClick={this.componentDidMount}>Change image</button>
				{
					this.state.urlPic ?
						(<img src={this.state.urlPic} alt="pic"></img>) : null
				}
      </div>
    );
  }
}

export default DogApp;
