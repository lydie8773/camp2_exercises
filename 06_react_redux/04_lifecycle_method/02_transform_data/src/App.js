import React, { Component } from 'react';
import './App.css';


// function getUserById(id, list){
// 	let userName;
// 	list.forEach((row) => {
// 		if(row.id === id){
// 			userName = row.name;
// 		}
// 	})
// 	return userName;
// }
//
// function Payment(props) {
// 	props.expenses.map((row, index) => {
// 		return (
// 			<table style="width:100%">
// 				<tr>
// 					<th>what has been paid</th>
// 					<th>by whom?</th>
// 				</tr>
// 				<tr key={index}>
// 					<td>{row.what}</td>
// 					<td>{this.getUserById(row.who)}</td>
// 				</tr>
// 			</table>
// 		)
// 	})
// }

// singleRowValue(row){
// 	return(
// 		Object.values(row).map((ele) => {
// 			return <td>{ele}</td>
// 		})
// 	);
// }
//
// tableValue(tableObject){
// 	return(
// 		tableObject.map((row) => {
// 			return <tr key={Object.values(row)[0]}>{this.singleRowValue(row)}</tr>
// 		})
// 	);
// }

// function Payment(props) {
// 	let userName;
//
// 	props.expenses.map((eleExpense) => {
// 		props.users.forEach((eleUsers) => {
// 			if (eleExpense.who === eleUsers.id){
// 				userName = eleUsers.name;
// 			}
// 		});
//
// 		return (
// 			<li>{eleExpense.what}, {userName}</li>
// 		)
// 	})
// }

// App has `expenses` and `users` as props
class App extends Component {
	// return (
	// 	<div className="App">
	// 		<Payment expenses={this.props.expenses} users={this.props.users} />
	// 	</div>
	// );

	// let userName;
	//
	// this.props.expenses.map((eleExpense) => {
	// 	this.props.users.forEach((eleUsers) => {
	// 		if (eleExpense.who === eleUsers.id) {
	// 			userName = eleUsers.name;
	// 		}
	// 	});
	//
	//
	// 	return (
	// 		<div className="App">
	// 			<li>{eleExpense.what}, {userName}</li>
	// 		</div>
	// 	)
	// })

	findUserById(id, list){
		let userName = " ";
		list.forEach((row) => {
			if(row.id === id){
				userName = row.name;
			}
		})
		return userName? userName:"Nobody";
	}

	payment(props){
		let listPayment = [];
		let json = {};
		props.expenses.forEach((ele) => {
			json = {"what": ele.what, "who": this.findUserById(ele.who, props.users)};
			listPayment.push(json);
		})
		return (
			listPayment.map((row, index) => {
				return (
					<tr key={index}>
						<td>{row.what}</td>
						<td>{row.who}</td>
					</tr>
				)
			})
		);
	}

	render(){
		return (
			<div className="container">
				<table className="table table-striped table-bordered table-responsive-md table-condensed datatable">
					<thead className="thead-dark">
						<tr>
							<th>what has been paid</th>
							<th>by whom?</th>
						</tr>
					</thead>
					<tbody>
						{this.payment(this.props)}
					</tbody>
				</table>

			</div>
		);
	}
}



export default App;
