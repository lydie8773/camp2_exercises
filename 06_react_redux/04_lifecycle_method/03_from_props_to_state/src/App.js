import React, { Component } from 'react';
import './App.css';
import Book from "./Book"

class App extends Component {
  constructor(props) {
    super(props);
    this.state = {
      isbn: "978059651774",
    }
  }
  handleChange = (event) => {
    this.setState({isbn: event.target.value});
  }
  handleSearch = () => {
		this.setState({isbn: this.state.isbn});
  }
  render() {
    return (
      <div className="App">
        <input type="text" value={this.state.isbn} onChange={this.handleChange} />
        <button onClick={this.handleSearch}>Search</button>
        <Book isbn={this.state.isbn} />
      </div>
    );
  }
}

export default App;
