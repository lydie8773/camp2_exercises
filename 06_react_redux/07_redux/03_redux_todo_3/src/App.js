import React, { Component } from 'react';
import './App.css';
import Todo from "./modules/Todo"

class App extends Component {
  render() {
    return (
      <div className="App">
        <header className="App-header">

        </header>
        <Todo />
      </div>
    );
  }
}


export default App;
