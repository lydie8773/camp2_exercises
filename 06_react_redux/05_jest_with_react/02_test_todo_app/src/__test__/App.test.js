import React from 'react';
import renderer from 'react-test-renderer';
import App from "../App";
import {shallow, configure} from "enzyme";


import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });


describe("TodoList test ", () => {
	it("todo snapshots test root node ", () => {
		const component = renderer.create(<App />);
		let snapshot = component.toJSON();
		expect(snapshot).toMatchSnapshot();
	});

	it("snapshots: test first node of function addtask", () => {
		const component = renderer.create(<App />);
		let snapshot = component.toJSON();
		snapshot = snapshot.children[0].children[0].props.onSubmit;

		expect(snapshot).toMatchSnapshot();
	});

	it("Enzyme: add one task", () => {
		const component = shallow(<App />);
		// console.log("form: ", components.find("form").first().props());
		component.setState({newTask: "test New Task"});
		component.find("form").first().simulate("submit", { preventDefault() {} });

		expect(component.find("span").text()).toEqual("test New Task");
	});

	it("Enzyme: add more task when click button add", () => {
		const component = shallow(<App />);
		component.setState({newTask: "1 task"});
		component.find("form").first().simulate("submit", { preventDefault() {} });
		expect(component.find("li").length).toBe(1);
		component.setState({newTask: "2 task"});
		component.find("form").first().simulate("submit", { preventDefault() {} });
		expect(component.find("li").length).toBe(2);
		component.setState({newTask: "3 task"});
		component.find("form").first().simulate("submit", { preventDefault() {} });
		expect(component.find("li").length).toBe(3);
	});

	it("Enzyme: test checkbox when button delete appear", () => {
		const component = shallow(<App />);
		component.setState({newTask: "1 task"});
		component.find("form").first().simulate("submit", { preventDefault() {} });
		// console.log("checkobx: ", components.find("input").at(1).props());

		component.find("input").at(1).simulate("change");
		// console.log("checkobx: ", components.find("input").at(1).props());
		// console.log("button: ", components.find("button").props());
		// console.log(components.find("li").at(0).props());
		expect(component.find("input").length).toBe(3);
	});
})