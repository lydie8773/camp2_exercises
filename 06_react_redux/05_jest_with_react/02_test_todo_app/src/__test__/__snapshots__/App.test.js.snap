// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoList test  snapshots: test first node of function addtask 1`] = `[Function]`;

exports[`TodoList test  todo snapshots test root node  1`] = `
<div
  className="TodoList"
>
  <div
    className="new-todo"
  >
    <form
      onSubmit={[Function]}
    >
      <input
        onChange={[Function]}
        type="text"
        value=""
      />
      <button
        type="submit"
      >
        Add
      </button>
    </form>
  </div>
  <ul
    className="todos"
  />
</div>
`;
